### PR TITLE
Remove npm

### PR DESCRIPTION
**Pull request by mellery451**
Eliminate npm tests and remove test directory. The second commit is just a delete, so only 7dd5eda needs to be reviewed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,10 +16,6 @@
     "ripple-lib": "0.13.0-rc6.0",
     "simple-jsonrpc": "~0.0.2"
   },
-  "scripts": {
-    "pretest": "node test/pretest.js",
-    "test": "mocha test/websocket-test.js test/server-test.js test/*-test.{js,coffee}"
-  },
```
Comment (ximinez) [inline package.json]: I *think* this file is only used in the context of `npm`, so it can be deleted, too.

Comment (mellery451) [inline package.json]: will remove

**Comment by ximinez** [comment]
@mellery451 Here is the fix for the Appveyor build https://github.com/ripple/rippled/commit/fcd38dba8013f522338bda418104224ba332085d

File: src/test/shamap/SHAMapSync_test.cpp [hunk 2]
```diff
@@ -194,7 +194,7 @@ class sync_test : public beast::unit_test::suite
 
         BEAST_EXPECT(source.deepCompare (destination));
 
-        std::cerr << "Checking destination invariants" << std::endl;
```
Comment (vinniefalco) [inline src/test/shamap/SHAMapSync_test.cpp]: Are there any other uses of `std::cerr` in the code base?

**Comment by vinniefalco** [comment]
Spelling: "Apveyor" in commit message

**Comment by ximinez** [comment]
Ugh. I'm all in favor of squashing that commit down, though, so it doesn't _need_ to be fixed.

File: src/test/shamap/SHAMapSync_test.cpp (hunk 2 above)
Comment (mellery451) [inline src/test/shamap/SHAMapSync_test.cpp]: yeah - I see about 30 hits for `::cerr` in src/ripple

**Comment by mellery451** [comment]
@ximinez -- will do

**Comment by vinniefalco** [comment]
@ximinez Oh... true.

File: Builds/ArchLinux/PKGBUILD [hunk 3]
```diff
@@ -25,9 +25,6 @@ build() {
 }
 
 check() {
-	cd "$srcdir/$pkgname"
```
Comment (ximinez) [inline Builds/ArchLinux/PKGBUILD]: Will the unittests still work without changing dir?

Comment (mellery451) [inline Builds/ArchLinux/PKGBUILD]: seems likely that it would work without the chdir...@wilsonianb, how can I vet changes to PKGBUILD ?

Comment (wilsonianb) [inline Builds/ArchLinux/PKGBUILD]: I'm not familiar with Arch Linux / PKGBUILD, but I think you are right to keep the `cd`

**Comment by wilsonianb** [comment]
This comment can be removed
https://github.com/mellery451/rippled/blob/remove_npm/appveyor.yml#L116

**Comment by wilsonianb** [comment]
Dependency can be removed
https://github.com/mellery451/rippled/blob/2eaf5019d8ed7001d8f1b9c8d5ee8980927e98e5/Builds/ArchLinux/PKGBUILD#L12

**Comment by wilsonianb** [comment]
All `nodejs`s can be removed from here:
https://github.com/mellery451/rippled/blob/2eaf5019d8ed7001d8f1b9c8d5ee8980927e98e5/Builds/Ubuntu/install_rippled_depends_ubuntu.sh#L28
https://github.com/mellery451/rippled/blob/2eaf5019d8ed7001d8f1b9c8d5ee8980927e98e5/Builds/Ubuntu/install_rippled_depends_ubuntu.sh#L41
https://github.com/mellery451/rippled/blob/2eaf5019d8ed7001d8f1b9c8d5ee8980927e98e5/Builds/Ubuntu/install_rippled_depends_ubuntu.sh#L50

**Comment by mellery451** [comment]
@wilsonianb - removed those nodejs mentions...thanks.

**Comment by wilsonianb** [comment]
:+1:

**Comment by nbougalis** [comment]
Merged as e3ff30657cab64b1c16cfb9989bcac7c93572bd9. And there was much rejoicing.

**Pull request closed without merging**